DoSomethingArchive/gambit-content
# SMS game creation

#### What's this PR do?

Implements `SGCompetitiveStoryController.createGame` to save the game and user documents to the database. With those in persistent storage, we can later reference them to figure out and update a user's state in the game. At the end, it also opts in the alpha and beta to their respective paths.

The `sgCompetitiveStory` schema was updated a to reflect how it'll be used to just track the current progression of players through the story instead of defining the story paths. The responsibility now falls on `competitive-stories.json` which has been populated with the start of a test story.

Finally, `mobilecommons.optin` was updated to handle multiple beta phone numbers.
#### Where should the reviewer start?

`SGCompetitiveStoryController.js` for the game creation and Mobile Commons opt ins. Then `mobilecommons.js` for the changes that handle multiple beta phone numbers.
#### How should this be manually tested?

I've successfully tested locally. If you want to too...

1) Make sure you've got Mongo running on your local
2) Start up the web service: `node server.js`
3) POST to http://localhost:4711/game/create with parameters:

```
type=competitive-story
story_id=1
person[first_name]=<your name>
person[phone]=<your phone number>
friends[0][first_name]=friend0
friends[1][first_name]=friend1
friends[2][first_name]=friend2
friends[0][phone]=13015550101
friends[1][phone]=13015550102
friends[2][phone]=13015550103
```
#### What are the relevant tickets?

Closes #34 
Closes #35 
Builds towards #38
